pkg/auth/dependency/principal/password/login_id.go

```diff
@@ -38,6 +40,10 @@ func (c defaultLoginIDChecker) validate(loginIDs []LoginID) error {
 			return errors.New("login ID is empty")
 		}
 		amounts[loginID.Key]++
+
+		if err := c.validateLoginIDFormat(loginID); err != nil {
```

Comment:
In this validate function, some errors are programming errors, such as login ID is empty while some are user errors, such as invalid format. I think we want to tell the user and the developer that

1. Login ID exceeds maximum
2. Login ID is in invalid format

Other errors should be programming errors.

Reply:
I think except when login ID key is not allowed in configuration, other errors should be reported as validation errors to be consistent with schema validation of other APIs.